crates/astria-sequencer/src/transaction/state_ext.rs

```diff
@@ -55,51 +53,23 @@ pub(crate) trait StateWriteExt: StateWrite {
     }
 
     #[instrument(skip_all)]
-    fn put_transaction_index_of_action(&mut self, index: u64) {
-        self.nonverifiable_put_raw(
-            index_of_action_storage_key().as_bytes().to_vec(),
-            borsh::to_vec(&index).expect("serialize index of action"),
-        );
-    }
-
-    #[instrument(skip_all)]
-    async fn increment_transaction_index_of_action(&mut self) -> Result<()> {
-        let index = self
-            .get_transaction_index_of_action()
-            .await?
-            .expect("index of action should be set before incrementing it");
-        let index = index.checked_add(1).expect("increment index of action");
-        self.nonverifiable_put_raw(
-            index_of_action_storage_key().as_bytes().to_vec(),
-            borsh::to_vec(&index).expect("serialize index of action"),
-        );
+    fn increment_position_in_source_transaction(&mut self) -> Result<()> {
```

Comment:
Don't increment, but instead provide a method `set_index_of_current_action` (or something like that).